### PR TITLE
fix(jottacloud): purge a trashed folder via its original path (#397)

### PR DESCRIPTION
**Pull request by axpnet**
## What this fixes

Tracker #701 item 9 said permanent purge of a trashed folder is impossible through the Jottacloud API, and the #703 test fixtures could only be cleared from the web UI. That was measured on the Trash view only. rclone's backend, which this implementation follows, sends its hard delete as `?rmDir=true` on the ORIGINAL path, and that form was never tried on a tombstone. Measured on the test account on 2026-09-01 with the owner's clearance:

| request | answer | trash re-listed |
|---|---|---|
| `POST <jfs>/<user>/Jotta/Archive/<folder>?rmDir=true` on a trashed folder's original path | 200 | folder gone (done on two fixtures) |
| `?rm=true` (file form) at a folder | 404 | unchanged |
| `POST api.jottacloud.com/files/v1/purge_trash`, rclone's `cleanup` | 200 `{"files":0,"folders":2}` | empty |

So item 9 was "not addressable through the surface we probed", not "impossible", and this PR makes both operations real.

## The change

- **Permanent delete of a trashed folder** goes through the hard-delete form on the original path, after that path has been read back and confirmed to be a tombstone. The guard is not optional: the same POST at a LIVE folder destroys it, so a live folder at that path is refused with a message that says so, and a path that answers neither in the trash nor at its origin is `NotFound`. Files keep the Trash-view form that already works.
- **Empty trash** for Jottacloud: `empty_trash()` on the provider, a `jottacloud_empty_trash` command, and the amber button plus confirmation in the trash manager, following the Yandex manager. The success line carries the counts the server reports, not the selection.
- The #703 end-to-end test now purges its own fixture and asserts the bin is clean, instead of logging that it cannot.

## Verified

- Unit test: the purge URL targets `/Jotta/Archive/<name>?rmDir=true` and never the Trash view; the decision accepts a tombstoned root folder and refuses a live folder (naming it LIVE), a file, and garbage.
- Live, `live_purge_trashed_folder_and_guard` on the saved Jotta profile: a fresh folder with a subfolder is trashed and purged through the original-path form and leaves the bin; the same call at a live folder is refused with the LIVE message and the folder survives; `empty_trash` on the remaining fixture reports 0 files and 1 folder and the bin re-lists empty. The #703 test cleanup now purges its fixture instead of logging that it cannot.
- `cargo fmt`, `cargo clippy --lib --tests -D warnings`, `cargo test --lib providers::jottacloud`, `tsc --noEmit` (7.0.2), all clean.

## Follow-ups this measurement opened, not in this PR

Tracker #701 item 9 is rewritten from this measurement, and the #397 reply that repeated it is corrected. Two more findings from the same session go to their own PR: the CLI and the GUI keep two different Jottacloud refresh chains on one station (`jottacloud_refresh` versus `jottacloud_refresh_<id>`), and the profile export prefers the per-profile chain even when it is the stale one, which is why an exported profile could arrive dead on the other station tonight.

## Summary by CodeRabbit

- **New Features**
  - Added an “Empty Trash” option to the Jottacloud trash manager.
  - Added a confirmation prompt before permanently deleting all trashed items.
  - Jottacloud trash cleanup now reports deleted file and folder counts.
  - Trash and file listings refresh after emptying the bin.
  - Added loading protection and error feedback during cleanup.

- **Bug Fixes**
  - Improved trashed-folder handling to prevent accidental deletion of active folders.
  - Improved confirmation behavior when dismissing dialogs or pressing Escape.

### CODE REVIEW

**Comment by axpnet** [comment]
@coderabbitai full review

**Comment by axpnet** [comment]
@coderabbitai full review

**Comment by axpnet** [comment]
@coderabbitai full review

**Comment by axpnet** [comment]
@coderabbitai full review

**Comment by axpnet** [comment]
@coderabbitai full review

**Comment by axpnet** [comment]
@coderabbitai full review

**Comment by axpnet** [comment]
@coderabbitai full review

File: docs/COMMAND-INVENTORY.json [hunk 1]
```diff
@@ -947,7 +947,7 @@
     ]
   },
   "parity": {
-    "name_matched_count": 479,
+    "name_matched_count": 480,
```
[the remark this answers is not in the document]
Comment (axpnet) [reply docs/COMMAND-INVENTORY.json:950]: Not taken, measured instead: the snapshot is generated by the CLI itself, and its parity pass matched jottacloud_empty_trash by token set to an existing command, so the regenerated file says name_matched_count 480 and name_unmatched_count 371, which sums to the 851 Tauri commands. Editing the counts by hand to 479/372 would contradict the generator and fail the self-consistency test.

File: src-tauri/src/providers/jottacloud.rs [hunk 2]
```diff
@@ -2648,21 +2684,49 @@ impl JottacloudProvider {
...
+                clean, reason
+            )));
         }
 
+        let dir_url = self.purge_trashed_dir_url(clean);
+        jotta_log(&format!(
+            "Purging trashed folder via original path: {}",
+            dir_url
+        ));
+        let resp = self.post_command_with_retry(&dir_url).await?;
```
[the remark this answers is not in the document]
Comment (axpnet) [reply src-tauri/src/providers/jottacloud.rs:2771]: Partly taken, and the rest declared: JFS offers no conditional delete (no revision or ETag precondition on rmDir), so the window between the tombstone GET and the POST is one round trip and cannot be closed from the client; that limit is now written in the method's doc, and the check keeps refusing the ordinary case, a live folder that was never the one selected. Not exposing folder purge at all would leave users with the web UI only, which is what item 9 of the tracker wrongly claimed. Commit on this branch.

File: src-tauri/src/providers/jottacloud.rs [hunk 3]
```diff
@@ -2672,11 +2736,49 @@ impl JottacloudProvider {
...
+        if !status.is_success() {
+            return Err(ProviderError::ServerError(format!(
+                "Empty trash failed ({}): {}",
+                status,
+                sanitize_api_error(&body)
+            )));
+        }
+        let counts: serde_json::Value = serde_json::from_str(&body).unwrap_or_default();
+        let files = counts["files"].as_u64().unwrap_or(0);
+        let folders = counts["folders"].as_u64().unwrap_or(0);
```
[the remark this answers is not in the document]
Comment (axpnet) [reply src-tauri/src/providers/jottacloud.rs]: Taken: parse_purge_trash_counts deserialises the two u64 fields and returns ParseError on non-JSON, a missing field, a string or a negative number; a unit test covers each shape, and the manager no longer logs a silent zero as a confirmed count. Commit on this branch.

File: src-tauri/src/providers/jottacloud.rs [hunk 4]
```diff
@@ -3747,27 +3880,153 @@ mod tests {
...
+
+    // ─── Live check of the purge guard and of empty_trash (#397, item 9) ──
+    //
+    // Run explicitly (never in CI):
+    //   cargo test --release --lib live_purge_trashed_folder_and_guard -- --ignored --nocapture
+    // Env: JOTTA_TEST_PROFILE (default "Jotta"). Test account only: the last
+    // step empties the WHOLE trash of that account.
+    #[tokio::test]
+    #[ignore = "live Jottacloud test; run explicitly"]
+    async fn live_purge_trashed_folder_and_guard() {
```
[the remark this answers is not in the document]
Comment (axpnet) [reply src-tauri/src/providers/jottacloud.rs:4061]: Taken: the whole-bin step now runs only with JOTTA_TEST_ALLOW_EMPTY_TRASH=1 on top of the ignored marker; without it the second fixture is purged on its own and the step is reported as skipped. Commit on this branch.

File: src/components/JottacloudTrashManager.tsx [hunk 5]
```diff
@@ -255,6 +287,30 @@ export function JottacloudTrashManager({ onClose, onRefreshFiles }: JottacloudTr
           </div>
         </div>
       )}
+
+      {pendingEmptyConfirm && (
+        <div className="fixed inset-0 z-[10000] bg-black/50 flex items-center justify-center" role="dialog" aria-modal="true" onClick={() => setPendingEmptyConfirm(false)}>
```
[the remark this answers is not in the document]
Comment (axpnet) [reply src/components/JottacloudTrashManager.tsx]: Taken, for both confirmation dialogs in this manager (the permanent-delete one had the same bubbling): the backdrop click stops propagation before dismissing the confirmation, so it no longer closes the whole trash manager. Commit on this branch.

File: src/components/JottacloudTrashManager.tsx [hunk 6]
```diff
@@ -255,6 +287,30 @@ export function JottacloudTrashManager({ onClose, onRefreshFiles }: JottacloudTr
...
+            </p>
+            <div className="flex justify-end gap-2">
+              <button
+                onClick={() => setPendingEmptyConfirm(false)}
+                className="px-4 py-2 text-gray-600 dark:text-gray-400 hover:bg-gray-100 dark:hover:bg-gray-700 rounded-lg"
+              >
+                {t('common.cancel')}
+              </button>
+              <button
+                onClick={confirmEmptyTrash}
```
[the remark this answers is not in the document]
Comment (axpnet) [reply src/components/JottacloudTrashManager.tsx:312]: Taken: the confirm button is disabled while an action runs and confirmEmptyTrash returns early if one is already in flight. Commit on this branch.

File: src-tauri/src/providers/jottacloud.rs [hunk 7]
```diff
@@ -3562,6 +3695,55 @@ mod tests {
                 "{name} mirrors a credential and never unmirrors it: a raw delete leaves the partition copy alive"
             );
         }
+    /// #397 / tracker #701 item 9: a trashed folder is purged with the
```
[the remark this answers is not in the document]
Comment (axpnet) [reply src-tauri/src/providers/jottacloud.rs:3700]: Already closed on the current head 5b7f740d: that brace was lost by the automatic conflict resolution during the rebase onto main and restored in the same rebase before the push you see now (fmt, clippy and the 27 Jottacloud unit tests, both new ones included, run green on 5b7f740d). The head you reviewed, 4ff98085, was live for a few minutes and is gone.

File: src/components/JottacloudTrashManager.tsx [hunk 8]
```diff
@@ -112,6 +113,28 @@ export function JottacloudTrashManager({ onClose, onRefreshFiles }: JottacloudTr
...
+      humanLog.updateEntry(logId, { status: 'error', message: `[Jottacloud] Failed to empty trash` });
+      setError(String(err));
+    } finally {
+      setActionLoading(null);
+    }
+  };
+
   useEffect(() => {
     const handleKey = (e: KeyboardEvent) => {
       if (e.key === 'Escape') onClose();
```
[the remark this answers is not in the document]
Comment (axpnet) [reply src/components/JottacloudTrashManager.tsx]: Taken: Escape now dismisses the innermost surface first, the pending empty-trash or delete confirmation, and closes the manager only when neither is open; both states are in the effect's dependencies. Commit on this branch.
